src/status_im/ui/screens/wallet/send/events.cljs

```diff
@@ -46,53 +66,93 @@
+      (dispatch-transaction-completed result))))
 
+;;TRANSACTION QUEUED signal from status-go
 (handlers/register-handler-fx
-  ::transaction-modal-completed
-  (fn [{db :db} [_ {:keys [id response]}]]
+  :transaction-queued
+  [(re-frame/inject-cofx :now)]
+  (fn [{:keys [db now]} [_ {:keys [id message_id args] :as transaction}]]
+    (if (transaction-valid? transaction)
+      (let [{:keys [from to value data gas gasPrice]} args
+            ;;TODO (andrey) revisit this map later (this map from old transactions, idk if we need all these fields)
+            transaction {:id         id
+                         :from       from
+                         :to         to
+                         :value      (money/bignumber value)
+                         :data       data
+                         :gas        (money/to-decimal gas)
+                         :gas-price  (money/to-decimal gasPrice)
+                         :timestamp  now
+                         :message-id message_id}
+            sending-from-chat? (not (get-in db [:wallet :send-transaction :waiting-signal?]))
+            new-db (assoc-in db [:wallet :transactions-unsigned id] transaction)
+            sending-db {:id         id
+                        :from-chat? sending-from-chat?}]
+        (if sending-from-chat?
+          ;;SENDING FROM CHAT
+          {:db       (assoc-in new-db [:wallet :send-transaction] sending-db) ; we need to completly reset sending state here
+           :dispatch [:navigate-to-modal :wallet-send-transaction-modal]}
+          ;;SEND SCREEN WAITING SIGNAL
+          (let [{:keys [later? password]} (get-in db [:wallet :send-transaction])
+                new-db' (update-in new-db [:wallet :send-transaction] merge sending-db)] ; just update sending state as we are in wallet flow
+            (if later?
+              ;;SIGN LATER
+              {:db                      (assoc-in new-db' [:wallet :send-transaction :waiting-signal?] false)
+               :dispatch                [:navigate-back]
+               ::show-transaction-moved nil}
+              ;;SIGN NOW
+              {:db                  new-db'
+               ::accept-transaction {:id           id
+                                     :password     password
+                                     :on-completed on-transactions-completed}}))))
+      {:discard-transaction id})))
+
+;;TRANSACTION FAILED signal from status-go
+(handlers/register-handler-fx
+  :transaction-failed
+  (fn [{{:keys [view-id modal] :as db} :db} [_ {:keys [id error_code error_message] :as event}]]
+    (let [send-transaction (get-in db [:wallet :send-transaction])]
+      (case error_code
+
+        ;;WRONG PASSWORD
+        constants/send-transaction-password-error-code
+        {:db (assoc-in db [:wallet :send-transaction :wrong-password?] true)}
+
+        ;;NO ERROR, DISCARDED, TIMEOUT or DEFAULT ERROR
+        (if (and (= (:id send-transaction) id)
```

Comment:
Extracting this logic as a function would improve readability.

Reply:
too many parameters, tried, readability not improved

Reply:
tried one more time, made it readable :)